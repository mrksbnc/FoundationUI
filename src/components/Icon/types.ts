export declare type IconObj = {
	name: string;
	path: string | null;
	svg: string | null;
};
