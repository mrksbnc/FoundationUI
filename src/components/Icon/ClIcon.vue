<template>
	<div :class="[`w-[${size}px] h-[${size}px]`]">
		<i :style="iconStyle" />
	</div>
</template>

<script setup lang="ts">
	import { Color } from '@/data';
	import { getIcon, Icon, type IconObj } from '.';
	import { toRefs, type PropType, computed, ref, onMounted, onBeforeMount, onUpdated } from 'vue';

	const props = defineProps({
		icon: {
			type: String as PropType<Icon>,
			required: true,
		},
		size: {
			type: Number,
			default: 24,
		},
		color: {
			type: String as PropType<Color | string>,
			default: Color.black,
		},
	});

	const { icon, size, color } = toRefs(props);

	const iconObj = ref<IconObj | null>(null);

	const computedIconSizeStyle = computed(() => {
		return {
			width: `${size.value}px`,
			height: `${size.value}px`,
		};
	});

	const computedIconColorStyle = computed(() => {
		return {
			fill: color.value,
		};
	});

	const iconStyle = computed(() => {
		return {
			...computedIconSizeStyle.value,
			...computedIconColorStyle.value,
			maskImage: `url(${iconObj.value?.default})`,
			webkitMaskImage: `url(${iconObj.value?.default})`,
			backgroundColor: color.value,
		};
	});

	onMounted(() => {
		iconObj.value = getIcon(icon.value);
	});

	onUpdated(() => {
		iconObj.value = getIcon(icon.value);
	});
</script>

<style scoped lang="scss">
	i {
		display: block;
		width: 100%;
		height: 100%;
		mask-size: contain;
		mask-repeat: no-repeat;
		-webkit-mask-size: contain;
		mask-position: center center;
		-webkit-mask-repeat: no-repeat;
		-webkit-mask-position: center center;
	}
</style>
