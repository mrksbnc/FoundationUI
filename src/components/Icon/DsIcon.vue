<template>
	<div :class="[`w-[${size}px] h-[${size}px]`]">
		<i :style="iconStyle" />
	</div>
</template>

<script setup lang="ts">
	import { getIcon, Icon, type IconObj } from '.';
	import { toRefs, type PropType, computed, ref, onMounted, onBeforeMount, onUpdated } from 'vue';

	const props = defineProps({
		icon: {
			type: String as PropType<Icon>,
			required: true,
		},
		size: {
			type: Number,
			default: 24,
		},
	});

	const { icon, size } = toRefs(props);

	const iconObj = ref<IconObj | null>(null);

	const iconStyle = computed(() => {
		return {
			width: `${size.value}px`,
			height: `${size.value}px`,
			maskImage: `url(${iconObj.value?.default})`,
			webkitMaskImage: `url(${iconObj.value?.default})`,
		};
	});

	onMounted(() => {
		iconObj.value = getIcon(icon.value);
	});

	onUpdated(() => {
		iconObj.value = getIcon(icon.value);
	});
</script>

<style scoped lang="scss">
	i {
		width: 100%;
		height: 100%;
		display: block;
		fill: currentColor;
		mask-size: contain;
		mask-repeat: no-repeat;
		background: currentColor;
		-webkit-mask-size: contain;
		mask-position: center center;
		-webkit-mask-repeat: no-repeat;
		-webkit-mask-position: center center;
	}
</style>
