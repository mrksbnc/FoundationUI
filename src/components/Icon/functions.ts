import { iconCollection, type Icon, type IconObj } from '.';

export function getIcon(key: Icon): IconObj | null {
	return iconCollection.get(key) ?? null;
}
