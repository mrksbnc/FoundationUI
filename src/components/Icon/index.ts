export { getIcon } from './functions';
export type { IconObj } from './types';
export { default as RuiIcon } from './RuiIcon.vue';
export { IconCollection, Icon } from './constants';
