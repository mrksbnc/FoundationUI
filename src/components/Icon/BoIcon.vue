<template>
	<i
		:id="id"
		role="img"
		:style="style"
		:aria-label="name"
		:class="['bo-icon', boIconClass]"
	/>
</template>

<script setup lang="ts">
import type { OptionalCss } from '@/types'
import type { Icon } from '@/components/Icon'
import { iconMap, IconSize } from '@/components/Icon'
import {
	type PropType,
	toRefs,
	computed,
	onUpdated,
	ref,
	onBeforeMount,
} from 'vue'

const props = defineProps({
	name: {
		type: String as PropType<Icon>,
		required: true,
	},
	size: {
		type: String as PropType<IconSize>,
		default: () => IconSize.M,
	},
	color: {
		type: String,
		default: () => null,
	},
})

const { name, size, color } = toRefs(props)

const index = ref<number>(0)
const id = ref<string>(`bo-icon-${index.value + 1}`)
const icon = ref<{ default: string } | null>(null)

const style = computed<Record<string, string>>(() => {
	const i = iconMap.get(name.value) ?? null
	const defaultColor = 'var(--icon-color)'

	return {
		'background-color': color.value ? color.value : defaultColor,
		maskImage: `url('${i?.default}')`,
		webkitMaskImage: `url('${i?.default}')`,
	}
})

const boIconClass = computed<OptionalCss>(() => {
	const classes: OptionalCss = {}

	switch (size.value) {
		case IconSize.XXS:
			classes['xxs'] = true
			break
		case IconSize.XS:
			classes['xs'] = true
			break
		case IconSize.S:
			classes['sm'] = true
			break
		case IconSize.M:
			classes['md'] = true
			break
		case IconSize.L:
			classes['lg'] = true
			break
		case IconSize.XL:
			classes['xl'] = true
			break
		case IconSize.XXL:
			classes['xxl'] = true
			break
	}

	return classes
})

onBeforeMount(() => {
	icon.value = iconMap.get(name.value) ?? null
})

onUpdated(() => {
	icon.value = iconMap.get(name.value) ?? null
})
</script>

<style scoped lang="scss">
@import url('./icon.scss');
</style>
