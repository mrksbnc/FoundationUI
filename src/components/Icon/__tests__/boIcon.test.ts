import { BoIcon, Icon, IconSize } from '@/components/Icon'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

describe('BoIcon', () => {
	it('renders properly', () => {
		const wrapper = mount(BoIcon, {
			props: {
				name: Icon.activity,
				size: IconSize.M,
			},
		})
		expect(wrapper).toMatchSnapshot()
	})
})
