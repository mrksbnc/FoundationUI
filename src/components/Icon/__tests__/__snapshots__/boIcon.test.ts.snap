// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoIcon > renders properly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <i
        aria-label="activity"
        class="bo-icon md"
        data-v-25c2865e=""
        id="bo-icon-1"
        role="img"
        style="mask-image: url('/src/assets/icon/activity.svg'); webkit-mask-image: url('/src/assets/icon/activity.svg');"
      />
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "attrsProxy": null,
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "slotsProxy": null,
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "attrsProxy": null,
          "bc": null,
          "bm": [
            [Function],
          ],
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "color": null,
            "name": "activity",
            "size": "m",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "boIconClass": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": {
                "md": true,
              },
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    Set {
                      [Circular],
                    },
                    Set {
                      ReactiveEffect {
                        "active": true,
                        "computed": ComputedRefImpl {
                          "__v_isReadonly": true,
                          "__v_isRef": true,
                          "_cacheable": true,
                          "_dirty": false,
                          "_setter": [Function],
                          "_value": {
                            "background-color": "var(--icon-color)",
                            "maskImage": "url('/src/assets/icon/activity.svg')",
                            "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
                          },
                          "dep": Set {
                            [Circular],
                          },
                          "effect": [Circular],
                        },
                        "deps": [
                          [Circular],
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                      [Circular],
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
            "color": ObjectRefImpl {
              "__v_isRef": true,
              "_defaultValue": undefined,
              "_key": "color",
              "_object": {
                "color": null,
                "name": "activity",
                "size": "m",
              },
            },
            "icon": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": {
                "default": "/src/assets/icon/activity.svg",
              },
              "_value": {
                "default": "/src/assets/icon/activity.svg",
              },
              "dep": undefined,
            },
            "id": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": "bo-icon-1",
              "_value": "bo-icon-1",
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    [Circular],
                    Set {
                      [Circular],
                    },
                    Set {
                      ReactiveEffect {
                        "active": true,
                        "computed": ComputedRefImpl {
                          "__v_isReadonly": true,
                          "__v_isRef": true,
                          "_cacheable": true,
                          "_dirty": false,
                          "_setter": [Function],
                          "_value": {
                            "background-color": "var(--icon-color)",
                            "maskImage": "url('/src/assets/icon/activity.svg')",
                            "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
                          },
                          "dep": Set {
                            [Circular],
                          },
                          "effect": [Circular],
                        },
                        "deps": [
                          [Circular],
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                      [Circular],
                    },
                    Set {
                      [Circular],
                    },
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
            },
            "index": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": 0,
              "_value": 0,
              "dep": undefined,
            },
            "name": ObjectRefImpl {
              "__v_isRef": true,
              "_defaultValue": undefined,
              "_key": "name",
              "_object": {
                "color": null,
                "name": "activity",
                "size": "m",
              },
            },
            "props": {
              "color": null,
              "name": "activity",
              "size": "m",
            },
            "size": ObjectRefImpl {
              "__v_isRef": true,
              "_defaultValue": undefined,
              "_key": "size",
              "_object": {
                "color": null,
                "name": "activity",
                "size": "m",
              },
            },
            "style": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": {
                "background-color": "var(--icon-color)",
                "maskImage": "url('/src/assets/icon/activity.svg')",
                "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
              },
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Set {
                      [Circular],
                    },
                    [Circular],
                    Set {
                      ReactiveEffect {
                        "active": true,
                        "computed": [Circular],
                        "deps": [
                          [Circular],
                          Set {
                            [Circular],
                          },
                        ],
                        "fn": [Function],
                        "parent": undefined,
                        "scheduler": [Function],
                      },
                      [Circular],
                    },
                    Set {
                      [Circular],
                    },
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [
                  Set {
                    [Circular],
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        Set {
                          [Circular],
                        },
                        Set {
                          [Circular],
                        },
                        [Circular],
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                ReactiveEffect {
                  "active": true,
                  "computed": ComputedRefImpl {
                    "__v_isReadonly": true,
                    "__v_isRef": true,
                    "_cacheable": true,
                    "_dirty": false,
                    "_setter": [Function],
                    "_value": {
                      "background-color": "var(--icon-color)",
                      "maskImage": "url('/src/assets/icon/activity.svg')",
                      "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
                    },
                    "dep": Set {
                      [Circular],
                    },
                    "effect": [Circular],
                  },
                  "deps": [
                    [Circular],
                    Set {
                      [Circular],
                    },
                  ],
                  "fn": [Function],
                  "parent": undefined,
                  "scheduler": [Function],
                },
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "color": null,
            "name": "activity",
            "size": "m",
          },
          "propsDefaults": {
            "color": null,
          },
          "propsOptions": [
            {
              "color": {
                "0": false,
                "1": true,
                "default": [Function],
                "type": [Function],
              },
              "name": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "size": {
                "0": false,
                "1": true,
                "default": [Function],
                "type": [Function],
              },
            },
            [
              "size",
              "color",
            ],
          ],
          "provides": {},
          "proxy": {
            "color": null,
            "name": "activity",
            "size": "m",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": {
                    "background-color": "var(--icon-color)",
                    "maskImage": "url('/src/assets/icon/activity.svg')",
                    "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
                  },
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        Set {
                          [Circular],
                        },
                        [Circular],
                        Set {
                          [Circular],
                          [Circular],
                        },
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": [
                  Set {
                    [Circular],
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        Set {
                          [Circular],
                        },
                        Set {
                          [Circular],
                        },
                        [Circular],
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": {
                    "md": true,
                  },
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        Set {
                          [Circular],
                        },
                        Set {
                          [Circular],
                        },
                        Set {
                          ReactiveEffect {
                            "active": true,
                            "computed": ComputedRefImpl {
                              "__v_isReadonly": true,
                              "__v_isRef": true,
                              "_cacheable": true,
                              "_dirty": false,
                              "_setter": [Function],
                              "_value": {
                                "background-color": "var(--icon-color)",
                                "maskImage": "url('/src/assets/icon/activity.svg')",
                                "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
                              },
                              "dep": Set {
                                [Circular],
                              },
                              "effect": [Circular],
                            },
                            "deps": [
                              [Circular],
                              Set {
                                [Circular],
                              },
                            ],
                            "fn": [Function],
                            "parent": undefined,
                            "scheduler": [Function],
                          },
                          [Circular],
                        },
                        [Circular],
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    ReactiveEffect {
                      "active": true,
                      "computed": ComputedRefImpl {
                        "__v_isReadonly": true,
                        "__v_isRef": true,
                        "_cacheable": true,
                        "_dirty": false,
                        "_setter": [Function],
                        "_value": {
                          "background-color": "var(--icon-color)",
                          "maskImage": "url('/src/assets/icon/activity.svg')",
                          "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
                        },
                        "dep": Set {
                          [Circular],
                        },
                        "effect": [Circular],
                      },
                      "deps": [
                        [Circular],
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "boIconClass": {
              "md": true,
            },
            "color": null,
            "icon": {
              "default": "/src/assets/icon/activity.svg",
            },
            "id": "bo-icon-1",
            "index": 0,
            "name": "activity",
            "props": {
              "color": null,
              "name": "activity",
              "size": "m",
            },
            "size": "m",
            "style": {
              "background-color": "var(--icon-color)",
              "maskImage": "url('/src/assets/icon/activity.svg')",
              "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
            },
          },
          "slots": {},
          "slotsProxy": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": null,
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [],
            "dynamicProps": [
              "id",
              "aria-label",
            ],
            "el": <i
              aria-label="activity"
              class="bo-icon md"
              data-v-25c2865e=""
              id="bo-icon-1"
              role="img"
              style="mask-image: url('/src/assets/icon/activity.svg'); webkit-mask-image: url('/src/assets/icon/activity.svg');"
            />,
            "key": null,
            "patchFlag": 14,
            "props": {
              "aria-label": "activity",
              "class": "bo-icon md",
              "id": "bo-icon-1",
              "role": "img",
              "style": {
                "background-color": "var(--icon-color)",
                "maskImage": "url('/src/assets/icon/activity.svg')",
                "webkitMaskImage": "url('/src/assets/icon/activity.svg')",
              },
            },
            "ref": null,
            "scopeId": "data-v-25c2865e",
            "shapeFlag": 1,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "i",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Volumes/Data/code/projects/bamboo/src/components/Icon/BoIcon.vue",
            "__name": "BoIcon",
            "__scopeId": "data-v-25c2865e",
            "components": {},
            "props": {
              "color": {
                "default": [Function],
                "type": [Function],
              },
              "name": {
                "required": true,
                "type": [Function],
              },
              "size": {
                "default": [Function],
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": [
            [Function],
          ],
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <i
          aria-label="activity"
          class="bo-icon md"
          data-v-25c2865e=""
          id="bo-icon-1"
          role="img"
          style="mask-image: url('/src/assets/icon/activity.svg'); webkit-mask-image: url('/src/assets/icon/activity.svg');"
        />,
        "key": null,
        "patchFlag": 0,
        "props": {
          "name": "activity",
          "ref": "VTU_COMPONENT",
          "size": "m",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Volumes/Data/code/projects/bamboo/src/components/Icon/BoIcon.vue",
          "__name": "BoIcon",
          "__scopeId": "data-v-25c2865e",
          "components": {},
          "props": {
            "color": {
              "default": [Function],
              "type": [Function],
            },
            "name": {
              "required": true,
              "type": [Function],
            },
            "size": {
              "default": [Function],
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <i
          aria-label="activity"
          class="bo-icon md"
          data-v-25c2865e=""
          id="bo-icon-1"
          role="img"
          style="mask-image: url('/src/assets/icon/activity.svg'); webkit-mask-image: url('/src/assets/icon/activity.svg');"
        />,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.3.8",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <i
    aria-label="activity"
    class="bo-icon md"
    data-v-25c2865e=""
    id="bo-icon-1"
    role="img"
    style="mask-image: url('/src/assets/icon/activity.svg'); webkit-mask-image: url('/src/assets/icon/activity.svg');"
  />,
}
`;
